server/cmwell-bg/src/main/scala/cmwell/bg/CMWellBGActor.scala

```diff
@@ -177,19 +179,23 @@ class CMWellBGActor(partition:Int, config:Config, irwService:IRWService, ftsServ
   }
 
   private def startImp = {
-    if(impStream == null) {
-      logger info "starting ImpStream"
-      impStream = new ImpStream(partition, config, irwService, zStore, ftsService, offsetsService, self)
-    } else
-      logger warn "requested to start Imp Stream but it is already running. doing nothing."
+    if(impOn) {
+      if (impStream == null) {
```

Comment:
can't we get along with a simple `lazy val impStream = new ImpStream(...)` instead of `var impStream: ImpStream = null` with this check?
It can potentially be dangerous with multi-threaded execution...

Reply:
this null is used in other places